src/map/mob.cpp

```diff
@@ -4208,6 +4201,80 @@ const std::string MobDatabase::getDefaultLocation() {
 	return std::string(db_path) + "/mob_db.yml";
 }
 
+template<typename T, size_t S> bool MobDatabase::parseDropNode(std::string nodeName, YAML::Node node, bool exists, uint8 max, T(&arr)[S]) {
+	const YAML::Node &dropNode = node[nodeName];
+	uint16 i = 0;
+
+	for (const YAML::Node &dropit : dropNode) {
+		if (i >= max) {
+			this->invalidWarning(dropit, "Maximum of %d monster %s met, skipping.\n", max, nodeName.c_str());
+			continue;
+		}
+
+		std::string item_name;
+
+		if (!this->asString(dropit, "Item", item_name))
+			return false;
+
+		item_data *item = itemdb_search_aegisname(item_name.c_str());
+
+		if (item == nullptr) {
+			this->invalidWarning(dropit["Item"], "Monster drop item %s does not exist, skipping.\n", item_name.c_str());
+			continue;
+		}
+
+		uint16 rate;
+
+		if (this->nodeExists(dropit, "Rate")) {
+			if (!this->asUInt16Rate(dropit, "Rate", rate))
+				return false;
+		} else {
+			if (!exists)
+				rate = 1;
+		}
+
+		bool steal = false;
+
+		if (this->nodeExists(dropit, "StealProtected")) {
+			if (!this->asBool(dropit, "StealProtected", steal))
+				return false;
+		}
+
+		uint16 group = 0;
+
+		if (this->nodeExists(dropit, "RandomOptionGroup")) {
+			std::string group_name;
+
+			if (!this->asString(dropit, "RandomOptionGroup", group_name))
+				return false;
+
+			if (!random_option_group.option_get_id(group_name.c_str(), group))
+				this->invalidWarning(dropit["RandomOptionGroup"], "Unknown random option group %s for monster drop, defaulting to no group.\n", group_name.c_str());
+		}
+
+		uint16 index;
+
+		if (this->nodeExists(dropit, "Index")) {
+			if (!this->asUInt16(dropit, "Index", index))
+				return false;
+
+			if (index > MAX_MVP_DROP - 1) {
```

Comment:
Could be me, but should this and below be "max" instead of the old hardcoded value?